### PR TITLE
spec-file: add numpy to require and cleanup

### PR DESCRIPTION
**Pull request by daveol**
Add numpy as a dependency for the python package since the rdm-tests now use numpy.
The rest of this commit is to solve rpmlint errors and warnings like:
 - unused-direct-shlib-dependency
 - postun-without-ldconfig
 - non-empty-%postun
 - and more

### CODE REVIEW

File: ola.spec.in [hunk 1]
```diff
@@ -1,10 +1,9 @@
-#missing el6 macros
-%if 0%{?el6}
```
Comment (peternewman) [inline ola.spec.in]: Does this remove support for el6 then?

File: ola.spec.in [hunk 2]
```diff
@@ -16,14 +15,13 @@ URL:		https://github.com/OpenLightingProject/ola
 Source:		https://github.com/OpenLightingProject/ola/releases/download/%{version}/%{name}-%{version}.tar.gz
 
 BuildRequires:	pkgconfig,pkgconfig(cppunit),pkgconfig(uuid),pkgconfig(avahi-client)
-BuildRequires:	pkgconfig(libusb-1.0) >= 1.0.2,pkgconfig(libmicrohttpd) >= 0.4.0
+BuildRequires:	pkgconfig(libusb-1.0) >= 1.0.2,pkgconfig(libmicrohttpd) >= 0.4.0%{!?el6:,pkgconfig(systemd)}
 BuildRequires:	libtool,bison,flex,python2-devel,openslp-devel,pkgconfig(libftdi%{!?el6:1}) >= 0.18
-BuildRequires:	protobuf-devel >= 0.2,protobuf-compiler,protobuf-python
+BuildRequires:	protobuf-devel >= 0.2,protobuf-compiler,protobuf-python,numpy
```
Comment (peternewman) [inline ola.spec.in]: I don't think the build currently requires it technically, as we don't run tests on that stuff yet, it's just the released package that needs it.

**Comment by peternewman** [comment]
Generally LGTM

File: ola.spec.in (hunk 1 above)
Comment (daveol) [inline ola.spec.in]: no, it just means the shims do their work whenever they need to and not only on el6

File: ola.spec.in (hunk 2 above)
Comment (daveol) [inline ola.spec.in]: I would rather let it be a buildrequire, in case we eventually do use them using ``` make all ``` or ``` make check ```

**Pull request merged**